### ISSUE TITLE
How to make sure sqlite3.dll is properly deployed when publishing to IIS?

### ISSUE THREAD

**Issue by Flavien** [aspnet/Microsoft.Data.Sqlite#118]
Right now when I try to run my website using the "IIS Express" command, or when deploying to Azure, I get an error because sqlite3.dll is not found:

    Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

How do I make sure sqlite3.dll is deployed, and in the right location?

**Comment by bricelam** [comment]
Are you using an ASP.NET 5 project? If not, are you using NuGet's `packages.config` or `project.json` workflow?

**Comment by Flavien** [comment]
Yes it's an ASP.NET 5 project.

**Comment by bricelam** [comment]
I think we're waiting on aspnet/dnx#2387 for this to work correctly.

**Comment by Flavien** [comment]
There is something strange going on: when hosted on IIS, it will only load sqlite3.dll if it is in `approot\packages\MyPackage\1.0.0\root\x86`, and it **won't work** if it is both in `approot\packages\MyPackage\1.0.0\root\x86` and `approot\packages\MyPackage\1.0.0\root`. However, when running with Kestrel, it doesn't work unless the file is in `approot\packages\MyPackage\1.0.0\root`.

Can we have some clarifications on where sqlite3.dll is expected to be placed (and is this behavior normal)?

**Comment by Flavien** [comment]
Actually on Kestrel, it does work if the file is only in `approot\packages\MyPackage\1.0.0\root\x86`, so I'll do that since that works on both. It is still strange that on IIS, having the file under root makes it not load the DLL.

**Comment by leak** [comment]
Same issue here running plain `dnx publish` to a local folder. The native dll is missing from the published package `approot\packages\Microsoft.Data.Sqlite\1.0.0-beta7`.

**Comment by bricelam** [comment]
@leak That sounds like a bug on DNX, could you [file a new issue on their repository](https://github.com/aspnet/dnx/issues/new)?

**Comment by bricelam** [comment]
Oh wait, @leak, that may have already been fixed in Beta 8 by aspnet/dnx#2387

**Comment by leak** [comment]
Just created a new `1.0.0-beta8-15618` project (is the mentioned fix in that version..?).

Still no sqlite3.dll in the published folders:
````
D:\testing>find . -iname *sqlite*
./approot/packages/Microsoft.Data.Sqlite
./approot/packages/Microsoft.Data.Sqlite/1.0.0-beta8-15572/lib/dnx451/Microsoft.Data.Sqlite.dll
./approot/packages/Microsoft.Data.Sqlite/1.0.0-beta8-15572/lib/dnxcore50/Microsoft.Data.Sqlite.dll
./approot/packages/Microsoft.Data.Sqlite/1.0.0-beta8-15572/Microsoft.Data.Sqlite.nuspec
````

**Comment by bricelam** [comment]
@leak File a new issue on DNX. It must not be implemented yet.

**Comment by leak** [comment]
Issue filed here: https://github.com/aspnet/dnx/issues/2741

**Comment by bricelam** [comment]
Closing since it looks like the DNX bug has traction and there isn't anything to do in this repo.

**Comment by NVentimiglia** [comment]
Is there a work around for this ? This and associated threads are marked closed without a clear work around. I am on RC2 and this issue only appears when running from a docker container from my mac. Works fine on windows and mac.

**Comment by natemcmaster** [comment]
@NVentimiglia this happens because your docker container doesn't have sqlite3 installed. See https://github.com/aspnet/Microsoft.Data.Sqlite/blob/1.0.0-rc2/README.md#requirements.

In the upcoming 1.0.0 release, you will no longer need this as we added a dependency on SQlite https://www.nuget.org/packages/sqlite/3.12.2 which will install `libsqlite3.so` to your app. See https://github.com/aspnet/Microsoft.Data.Sqlite/pull/250

**Comment by NVentimiglia** [comment]
@natemcmaster Perfect
